tests/ci.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Contributors to the Parsec project.
+# SPDX-License-Identifier: Apache-2.0
+
+set -euf -o pipefail
+
+# Points to Parsec's Unix Domain Socket on the CI
+export PARSEC_SERVICE_ENDPOINT="unix:/tmp/parsec.sock"
+
+#########
+# Build #
+#########
+RUST_BACKTRACE=1 cargo build
+
+#################
+# Static checks #
+#################
+# On native target clippy or fmt might not be available.
+if cargo fmt -h; then
+	cargo fmt --all -- --check
+fi
+if cargo clippy -h; then
+	cargo clippy --all-targets -- -D clippy::all -D clippy::cargo
+fi
+
+#############
+# CLI tests #
+#############
+./target/debug/parsec-tool --help
+./target/debug/parsec-tool ping
+./target/debug/parsec-tool list-authenticators
+./target/debug/parsec-tool list-providers
+./target/debug/parsec-tool list-opcodes -p 1
+./target/debug/parsec-tool create-ecc-key -k toto
+./target/debug/parsec-tool create-rsa-key -k tata
+./target/debug/parsec-tool list-keys
+./target/debug/parsec-tool export-public-key -k toto
+./target/debug/parsec-tool export-public-key -k tata
+./target/debug/parsec-tool generate-random -n 50
+./target/debug/parsec-tool sign -k toto "Les carottes sont cuites!"
```

Comment:
Am I too much of a prick if I ask you to pipe the output of these to some `test_blabla.out` files and then do a diff between that and the "correct" output in `test_blabla.valid` which you commit here?

Reply:
(I guess some of them can't be compared properly, e.g. the `generate-random`. Don't know if there's some way to at least check the format (and at least that it doesn't fail))

Reply:
YES

ok will do something like that, was trying to get away from it

Reply:
I added output checks for the commands that are somewhat stable. I guess ideally we should use `openssl` or other to check the decrypt/sign functions but we can maybe do that in subsequent PRs 😄 Also, we do test those methods in the Rust client and Parsec

Reply:
> we can maybe do that in subsequent PRs

Yep, agree!

> we do test those methods in the Rust client and Parsec

Well, yes and no - we don't check the PEM or ASN.1 conversions we've added